### PR TITLE
Communicators2

### PR DESCRIPTION
**Pull request by ElyesAhmed**
The opm simulators support a variable communicator instead of the fixed MPI_COMM_WORLD. A demo code has been added to main to test the use of non-global communicators. The compilation was done with and without USE_MPI and the tests with mpirun also.

### CODE REVIEW

**Comment by blattms** [comment]
Unfortunately, you will need to rebase this to resolve the conflicts. Would be nice if you would do that.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -32,6 +32,12 @@
 
 #include <fmt/format.h>
 
+using MPIComm = typename Dune::MPIHelper::MPICommunicator;
+#if DUNE_VERSION_NEWER(DUNE_COMMON, 2, 7)
+    using Communication = Dune::Communication<MPIComm>;
+#else
+    using Communication = Dune::CollectiveCommunication<MPIComm>;
+#endif
```
Comment (blattms) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:40]: Please don't do that in the global namespace. The name `Communication` is rather generic and could be used somewhere else (including 3rd party libraries). Maybe doing it in an Opm::Parallel namespace is better and would accomplish the same?

**Comment by ElyesAhmed** [comment]
OK!
________________________________

**Comment by atgeirr** [comment]
Superseded by #3571, closing.

**Pull request closed without merging**